From: Seth Heasley <seth.heasley@intel.com>
Date: 2013-02-06 15:08:00 -0800
Subject: Add Multifunction Device support for Intel Centerton SOC 
References: bnc#792793
Git-commit: 8ee3c2a79fe1df10bccd110d5b8cc13c5b9da709
Patch-mainline: v3.5

Adds DeviceIDs for the Intel Centerton SOC for Multifunction Device.

Acked-by: Jean Delvare <jdelvare@suse.de>
---
 drivers/mfd/lpc_sch.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

--- a/drivers/mfd/lpc_sch.c
+++ b/drivers/mfd/lpc_sch.c
@@ -36,6 +36,7 @@
 
 #define GPIOBASE	0x44
 #define GPIO_IO_SIZE	64
+#define GPIO_IO_SIZE_CENTERTON	128
 
 static struct resource smbus_sch_resource = {
 		.flags = IORESOURCE_IO,
@@ -62,6 +63,7 @@ static struct mfd_cell lpc_sch_cells[] =
 static struct pci_device_id lpc_sch_ids[] = {
 	{ PCI_DEVICE(PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_SCH_LPC) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_ITC_LPC) },
+	{ PCI_DEVICE(PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_CENTERTON_ILB) },
 	{ 0, }
 };
 MODULE_DEVICE_TABLE(pci, lpc_sch_ids);
@@ -99,7 +101,11 @@ static int __devinit lpc_sch_probe(struc
 	}
 
 	gpio_sch_resource.start = base_addr;
-	gpio_sch_resource.end = base_addr + GPIO_IO_SIZE - 1;
+
+	if(id->device == PCI_DEVICE_ID_INTEL_CENTERTON_ILB)
+		gpio_sch_resource.end = base_addr + GPIO_IO_SIZE_CENTERTON - 1;
+	else
+		gpio_sch_resource.end = base_addr + GPIO_IO_SIZE - 1;
 
 	for (i=0; i < ARRAY_SIZE(lpc_sch_cells); i++)
 		lpc_sch_cells[i].id = id->device;
